### DISCUSSION TITLE
STEP export translator version

### DISCUSSION THREAD

**Discussion by Kristoff6** [CadQuery/cadquery#1430] [Q&A] [answered]
Hello all,

This is my first post here. Thanks for this great project! After 2 weeks I'm able to completely migrate my CAD generation to CadQuery.

Now I'm facing an issue I don't understand. For one component some strange thing is happening. The generated STEP file did work before, but now it creates issues downstream.

Comparing working and non-working STEP files shows that they are identical, except:

- the Open CASCADE STEP translator version
- and of coarse the time stamp

The working file lists:
`#7 = PRODUCT('Open CASCADE STEP translator 7.7 13', 'Open CASCADE STEP translator 7.7 13','',(#8));`
while the non-working file lists: 
`#7 = PRODUCT('Open CASCADE STEP translator 7.7 2', 'Open CASCADE STEP translator 7.7 2','',(#8));`
So just a version difference...
Unfortunately, I can't share the the affected files as such. If needed, I'll try to reproduce the issue.

My questions:

- Is the version indication just informative info about the tranlator, or is it also used when importing this STEP file again? Based on what's happening, I assume the latter.
- How is this version change triggered in CadQuery?
- How can the translator version be controlled / specified?

Understanding this would help a lot! Thanks.

**Comment by jmwright** [comment]
It is hard to be sure without a minimum example that will reproduce the issues that you are seeing, or even knowing what the issues are, but I can compare translator version numbers with you.

I show...

 `#7 = PRODUCT('Open CASCADE STEP translator 7.7 1', 'Open CASCADE STEP translator 7.7 1','',(#8));`

...with the latest development version of CadQuery and OCP 7.7.1 installed from PyPI.

How are you installing CadQuery? If you used conda, you might want to make sure you did not get the dev package. You can run `conda list` to find which version you are using.  If you used pip, you can run `pip list` and look at the cadquery-ocp version.

**Reply by Kristoff6** [reply]
Thanks for the reply.

I understand, I'll try to reproduce the issue and share!

I did install CadQuery via pip as mentioned [here](https://cadquery.readthedocs.io/en/latest/installation.html). Current installation:

- cadquery 2.3.1
- cadquery-ocp 7.7.1

**Reply by Kristoff6** [reply]
[Part v7.7 2.step.zip](https://github.com/CadQuery/cadquery/files/13217138/Part.v7.7.2.step.zip)
[Part v7.7 13.step.zip](https://github.com/CadQuery/cadquery/files/13217139/Part.v7.7.13.step.zip)

Just remove the .zip expression, otherwise I couldnt upload them. Also, as I have no idea yet what drives this change, I just edited the 7.7 13 step file manually,

Searching previous files of this component that were generated, I find versions:

- 7.7 2
- 7.7 7
- 7.7 10
- 7.7 13

Of which 7.7 2 does something that creates issues downstream.

**Comment by jmwright** [comment]
The `7.7 2` file shows no errors for me when importing with FreeCAD and CadQuery. Can you double check that the shape is valid before exporting? https://github.com/CadQuery/cadquery/blob/8a9e5a09c92c6506a92eb2cd7365c3e40b8f60a7/cadquery/occ_impl/shapes.py#L568

Can you also describe what the process downstream is doing with the STEP file and what error it is giving?

**Reply by Kristoff6** [reply]
Ok, I'll use that check!

Downstream is a FEA step. What I of coarse can try is string replacing '7.7 2' to '7.7 13' before that.

**Reply by Kristoff6** [reply, marked as answer]
Seems like the number after 7.7 is a counter for the amount of exported step files in the CQ script and had nothing to do with the error after all.

The error was in Ansys. It could not correctly load a step file generated with a 360 revolve. I used 90° and 2x a mirror operation to fix it.

Thanks @jmwright for sparring on the issue.
